core/src/mender-artifact.c

```diff
@@ -179,6 +179,24 @@ mender_artifact_get_ctx(mender_artifact_ctx_t **ctx) {
     return MENDER_OK;
 }
 
+static bool
+is_compressed(const char *filename) {
+
+    /* Return true if artifact is compressed - otherwise false
+     */
+
+    const char  *compression_suffixes[]    = { ".gz", ".xz", ".zst" };
+    const size_t compression_suffixes_size = 3;
+
+    for (size_t i = 0; i < compression_suffixes_size; i++) {
```

Comment:
Consider:
- making it `static`. This way it does not get initialized every time you call the function
- making it `NULL`-terminated. This way you do not have to hard code or compute the size
- using `int`. It's usually the size that the CPU is optimized for
```suggestion
    static const char  *compression_suffixes[]    = { ".gz", ".xz", ".zst", NULL };

    for (int i = 0; NULL != compression_suffixes[i]; i++) {
```